packages/trader/src/Modules/Trading/Components/Form/TradeParams/Duration/duration-mobile.jsx

```diff
@@ -167,8 +167,10 @@ const Numbers = observer(
         };
 
         const setExpiryDate = (epoch, duration) => {
+            if (trade_duration_unit !== 'd') {
+                return moment.utc().add(Number(duration), 'days').format('D MMM YYYY, [23]:[59]:[59] [GMT +0]');
```

Comment:
Just a question: is it confirmed that we want to use this time for other trade types as well?

Reply:
Yes, Kate asked Harsh and he confirmed that for Rise/Fall, Higher/Lower, Touch/ No Touch Expiry is calculated the same way as for Vanillas